src/components/tl/nccl/tl_nccl_coll.c

```diff
@@ -64,6 +69,82 @@ static inline ucc_status_t ucc_nccl_check_dt_supported(ucc_datatype_t dt1,
     return UCC_OK;
 }
 
+ucc_tl_nccl_task_t * ucc_tl_nccl_init_task(ucc_base_coll_args_t *coll_args,
+                                           ucc_base_team_t *team)
+{
+    ucc_tl_nccl_context_t *nccl_ctx  = ucc_derived_of(team->context,
+                                                      ucc_tl_nccl_context_t);
+    ucc_tl_nccl_task_t    *task;
+    ucc_status_t           status;
+
+    task = ucc_mpool_get(&nccl_ctx->req_mp);
+    ucc_coll_task_init(&task->super, coll_args, team);
+    UCC_TL_NCCL_PROFILE_REQUEST_NEW(task, "tl_nccl_task", 0);
+    task->super.finalize           = ucc_tl_nccl_coll_finalize;
+    task->super.triggered_post     = ucc_tl_nccl_triggered_post;
+    task->completed                = NULL;
+    task->allgatherv_bcopy.scratch = NULL;
```

Comment:
i think this task->allgatherv_bcopy.scratch = NULL should be in allgather_bcopy_init. This is nccl generic task init, should not use alg specific union fields imho.